tests/e2e/script_test.go

```diff
@@ -279,7 +279,12 @@ func TestE2EScript(t *testing.T) {
 						existing := readObject(h, obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName())
 						resourceID, _, _ := unstructured.NestedString(existing.Object, "spec", "resourceID")
 						if resourceID == "" {
-							h.Fatalf("object did not have spec.resource: %v", existing)
+							externalRef, _, _ := unstructured.NestedString(existing.Object, "status", "externalRef")
```

Comment:
Not sure I follow why this is needed? Users should set `spec.resourceID` when acquiring the resource?